src/compile/data/stack.ts

```diff
@@ -180,27 +233,32 @@ export class StackNode extends DataFlowNode {
 
       transform.push({
         type: 'impute',
-        field: stackField,
+        field,
         groupby: stackby,
         key: dimensionField,
         method: 'value',
         value: 0
       });
     }
+    if(facetby) {
```

Comment:
In general, you should look at your overall PR diff and think why you do certain things. 
Here you add `if (facetby)` to wrap the main code for the transform assemble logic (as it creates the stack transform).  Having a case that we can skip the main logic is quite questionable. 

From your [earlier rationale](https://github.com/vega/vega-lite/pull/3771#discussion_r189981663), you say that facetby is `undefined`  in `makeFromTransform`.  But why should that be undefined? (That's actually something you set yourself.)

FYI, previously `facetby` was required but somehow you make it optional in this PR. (Why?)
I actually mentioned earlier that the facetby should be an empty array  `[]` in either case.  
Thus, facetby shouldn't be optional and then you don't have to add the if here. 

Just FYI, it's okay to forget some details I mentioned because this is quite a complicated code base.  But feel free to ask again if you are unsure about  something.  :)

A good case to test whether this logic is now correct for facetby is to try to see if you can modify 
https://github.com/vega/vega-lite/blob/master/examples/specs/normalized/trellis_stacked_bar_normalized.vl.json
such that you have the stack transform inside the inner `spec`.

```
{
  "$schema": "https://vega.github.io/schema/vega-lite/v2.json",
  "data": {
    "url": "data/barley.json"
  },
  "facet": {
    "column": {
      "field": "year",
      "type": "ordinal"
    }
  },
  "spec": {
    "transform": [{aggregate: ...}, {stack: ...}],
    "mark": "bar",
    "encoding": {
      ...
    }
  }
}



```

Reply:
Thanks for the review, Ham. I think now I understand where I was wrong. I made `facetby` a required field, and refactored `assemble`. 
I must have missed your comment about trying out `stack` inside a `faceted` spec. It's working now.
![screen shot 2018-05-23 at 12 01 30 am](https://user-images.githubusercontent.com/16068737/40382762-da16bad6-5e1c-11e8-8c14-b3f77e61d396.png)